src/main/java/duke/utilities/TaskList.java

```diff
@@ -0,0 +1,153 @@
+package duke.utilities;
+
+import duke.tasks.DeadlineTask;
+import duke.tasks.Task;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * The list of user tasks.
+ */
+public class TaskList {
+
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Instantiates a new empty {@code TaskList} object.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Instantiates a new {@code TaskList} object from the {@code taskList}
+     * provided.
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        tasks = taskList;
+    }
+
+    /**
+     * Adds the specified task to the end of the task list.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes the task at the specified position from the task list.
+     *
+     * @param index The index of the task to be deleted.
+     * @return The task that was deleted.
+     */
+    public Task deleteTask(int index) {
+        return tasks.remove(index);
+    }
+
+    /**
+     * Returns the task at the specified position in the task list.
+     *
+     * @param index The index of the task.
+     * @return The task at the position {@code index}.
+     */
+    public Task getTask(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * Marks the task at the specified position as done.
+     *
+     * @param index The index of the task to be marked as done.
+     * @return The task that was marked as done.
+     */
+    public Task markTaskAsDone(int index) {
+        Task task = tasks.get(index);
+        task.markDone();
+        return task;
+    }
+
+    /**
+     * Unmarks the task at the specified position as done.
+     *
+     * @param index The index of the task to be unmarked as done.
+     * @return The task that was unmarked as done.
+     */
+    public Task unmarkTaskAsDone(int index) {
+        Task task = tasks.get(index);
+        task.unmarkDone();
+        return task;
+    }
+
+    /**
+     * Returns all tasks that are due on the specified date.
+     *
+     * @param dueOnDate The date on which the tasks are due on.
+     * @return A list of tasks that are due on {@code dueOnDate}
+     */
```

Comment:
Great job in general for capitalising and punctuating all your javadoc comments!